domain/src/main/kotlin/no/nav/su/se/bakover/domain/Søknad.kt

```diff
@@ -5,7 +5,14 @@ import no.nav.su.se.bakover.common.now
 import java.util.UUID
 
 data class Søknad(
+    val sakId: UUID,
     val id: UUID = UUID.randomUUID(),
     val opprettet: Tidspunkt = now(),
-    val søknadInnhold: SøknadInnhold
+    val søknadInnhold: SøknadInnhold,
+    val søknadTrukket: SøknadTrukket? = null,
```

Comment:
Jeg lurer litt på denne, bare fordi vi vet at vi kommer til å få 2 lignende typer til (bortfalt og avvist). Kunne vi f.eks. kalle denne `SøknadAvsluttet` (avslutting er et samlebegrep fag har brukt) og så legge til en grunn/type i `SøknadAvsluttet` som sier hvilken type avslutting (trukket, bortfalt, avvist)? @hestad @jacob-meidell

Reply:
Med bortfalt tenker jeg på det tilfellet en saksbehandler har startet en behandling, men angrer og vil "fjerne" den. Jeg har foreløpig tenkt at en søknad ikke bortfaller, og heller er knyttet til behandlingen. Sånn som jeg forstår det skal man kunne opprette en ny søknadsbehandling dersom den forrige "bortfalt". Vil da bli det samme med revurderingsbehandling.

Når det kommer til avvist, så tenker jeg det er implementasjonsavhengig. Jeg vil slå et sterkt slag for å gi en HTTP 400 dersom man allerede har en ikke-trukket søknad med overlappende perioder. I så måte kan vi legge på disse dataene på saken, slik at frontend kan vise nåværende og forelda søknadsperioder, slik at veileder raskt kan se at her må brukeren trekke den forrige søknaden før en ny sendes inn. For den saks skyld bør jo personen som organiserer den fysiske "timen" til personen raskt kunne sjekke dette med nåværende løsning/tilgjengelig data.

Reply:
Det er nok også potensiale for at man feilregistrerer en papirsøknad f.eks., og ønsker å bortfalle søknaden.. Men jeg er usikker på hvor mye det skjer da. Sjekk med fag om bortfalling kun gjelder behandlinger, eller om det også gjelder søknader!
Forsåvidt også enig i at systemet kanskje burde kunne avvise automatisk, men det er nok noe vi burde snakke med fag om, eller om vi da også trenger mulighet for manuell avvisning.